graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from abc import ABCMeta, abstractmethod
+
+from frozendict import frozendict
+import six
+
+
+@six.python_2_unicode_compatible
+@six.add_metaclass(ABCMeta)
+class Statistics(object):
+    """Abstract class for statistics on classes and fields for all backend instances.
+
+    For the purposes of query cardinality estimation, we need statistics to provide better
+    cardinality estimates when operations like edge traversal or @filter directives are used.
+    All statistics except get_class_count() are optional, so if the statistic doesn't exist, a
```

Comment:
Based on this class, there are no statistics other than `get_class_count` -- nothing else is defined.

If there should be other statistics, I would define their methods and then make them return `None` as you say they should.

Reply:
If you define methods but then mark them `@abstractmethod` then they aren't truly optional. Did you try instantiating a class that doesn't implement those and relies on the default implementation of `return None`? You'll see that operation fail because of the `@abstractmethod`, because the default implementation is not usable.